lockbox-ios/Store/AutoLockStore.swift

```diff
@@ -98,13 +108,15 @@ extension AutoLockStore {
 
     private func setTimer(seconds: Int) {
         let timerValue = self.userDefaults.double(forKey: SettingKey.autoLockTimerDate.rawValue)
-        if timerValue != 0 && timerValue < Date().timeIntervalSince1970 {
+        if timerValue != 0 && timerValue > Date().timeIntervalSince1970 {
```

Comment:
* If `timerValue` is in the future, create a timer for that future date
* If `timerValue` is in the past, lock now
* Otherwise (assuming `timerValue` is not set or is **right now**) update the timer.

This seems to mean the timer is only updated if it's been cleared, or if `timerValue` happens to be the exact date/time.  It does not update from user interactions ...

Reply:
corrected about the first point (read `resetTimer()` as `setupTimer()` for the interaction Actions handling).